### ISSUE TITLE
HOW CAN I USE ST7735 DISPLAY WITH THE CODE

### ISSUE THREAD

**Issue by eydoooo** [daveruiz/doom-nano#12]
i cant use st7735 with the code and im a beginner so im not sure what to change in the code

**Comment by simplyrohan** [comment]
I'm currently working on it. I'll let you know if I get it to work

**Comment by simplyrohan** [comment]
https://github.com/simplyrohan/doom-nano

In `constants.h` set your your display to color or not. Also in `display.h` set the pins for your screen

**Comment by eydoooo** [comment]
Are these the only steps? I'm currently not home yet 
and i will work on it as soon as im home. Also Thanks for helping me.

> https://github.com/simplyrohan/doom-nano
> 
> In `constants.h` set your your display to color or not. Also in `display.h` set the pins for your screen

**Comment by simplyrohan** [comment]
You may need to make some hardware adjustments but they should mostly be in display.h and constants.h. Also, you will need to adjust for if you have INPUT_PULLUP buttons or not. Let me know if you have questions

**Comment by simplyrohan** [comment]
You can also check the commit that I made that contained all the changes I made.
[Commit](https://github.com/simplyrohan/doom-nano/commit/a4855fe76299c4fc3208d29e915a7bf56605dee7)

**Comment by eydoooo** [comment]
> You can also check the commit that I made that contained all the changes I made. [Commit](https://github.com/simplyrohan/doom-nano/commit/a4855fe76299c4fc3208d29e915a7bf56605dee7)

Thanks alot for helping me. I'm guessing the commit is the changed display.h/constant.h code?
I'm not really used to github so most of the things I don't know just yet.

**Comment by simplyrohan** [comment]
Yup, that's correct! But all of those use my display and hardware setup. Just change the pin definitions (they are all labeled) to your setup

**Comment by eydoooo** [comment]
> Yup, that's correct! But all of those use my display and hardware setup. Just change the pin definitions (they are all labeled) to your setup

Thanks man your a G

**Comment by simplyrohan** [comment]
Let me know if it works

**Comment by eydoooo** [comment]
> Let me know if it works

i get this error class Adafruit_SSD1306<128, 64>' has no member named 'fillScreen
im guessing its related to the display.h code but i cant find it.

**Comment by simplyrohan** [comment]
I think you are using the old display.h (in this repository). It seems you are only using my main script. To use all if my modifications use 
`git clone https://github.com/simplyrohan/doom-nano`

You will need to redo the changes though

**Comment by eydoooo** [comment]
> I think you are using the old display.h (in this repository). It seems you are only using my main script. To use all if my modifications use `git clone https://github.com/simplyrohan/doom-nano`
> 
> You will need to redo the changes though

im sorry i didnt quiet understand. can u please explain to me in monkey terms?😊


EDIT: ok so i opened the link u provided me (https://github.com/simplyrohan/doom-nano) and it took me to ur commit which i then downloaded the zip file and when i opened it it still gave me the same error.

**Comment by simplyrohan** [comment]
Hmm strange, I'll check it out tommorow

**Comment by simplyrohan** [comment]
Really sorry for the late reply

Could you list out what you did step by step to upload the code to the board. Can you also paste the error

I doubt that its the same error as my fork doesn't use the same display object

**Comment by miguehernandezc27-cpu** [comment]
Hi, sorry to bother you, but I have a question. I'm getting an error in constants.h regarding this part of the code: "constexpr uint8_t DATA_CLOCK = 11;
constexpr uint8_t DATA_LATCH = 12;
constexpr uint8_t DATA_SERIAL = 13;

// Sound
constexpr uint8_t SOUND_PIN = 1;"

Again, sorry to bother you, I'm new to this and I don't know how to fix it.

**Comment by miguehernandezc27-cpu** [comment]
And if you have time, could you help me, simplyrohan, please?

**Comment by miguehernandezc27-cpu** [comment]
This is the error I'm getting
Compilation error: redefinition of 'constexpr const uint8_t DATA_CLOCK'
